src/Microsoft.Performance.Toolkit.Engine.Tests/ProcessingOptionsResolverTests.cs

```diff
@@ -0,0 +1,254 @@
+﻿// Copyright (c) Microsoft Corporation.
+// Licensed under the MIT License.
+
+using Microsoft.Performance.SDK.Processing;
+using Microsoft.Performance.SDK.Processing.DataSourceGrouping;
+using Microsoft.Performance.SDK.Processing.Options;
+using Microsoft.Performance.SDK.Runtime;
+using Microsoft.Performance.Testing;
+using Microsoft.Performance.Toolkit.Engine;
+using Microsoft.Performance.Toolkit.Engine.Tests.TestCookers.Source123;
+using Microsoft.Performance.Toolkit.Engine.Tests.TestCookers.Source4;
+using Microsoft.Performance.Toolkit.Engine.Tests.TestCookers.Source5;
+using Microsoft.VisualStudio.TestTools.UnitTesting;
+using System;
+using System.Collections.Generic;
+using System.Linq;
+
+namespace Microsoft.Performance.SDK.Tests
+{
+    [TestClass]
+    public class ProcessingOptionsResolverTests
+    {
+        [TestMethod]
+        [UnitTest]
+        public void DefaultGlobalProcessingOptionsResolverTest()
+        {
+            var sut = new GlobalProcessingOptionsResolver(ProcessorOptions.Default);
+
+            Assert.IsNotNull(sut, "Options Resolver is null");
+
+            var dsg1 = new DataSourceGroup(
+                new[] { new FileDataSource("test1" + Source123DataSource.Extension) },
+                new DefaultProcessingMode());
+
+            var dsg2 = new DataSourceGroup(
+                new[] { new FileDataSource("test2" + Source4DataSource.Extension) },
+                new DefaultProcessingMode());
+
+            ProcessingSourceReference.TryCreateReference(typeof(Source123DataSource), out var psr123);
+            ProcessingSourceReference.TryCreateReference(typeof(Source4DataSource), out var psr4);
+
+            // Ensure these are default
+            Assert.AreEqual(ProcessorOptions.Default, sut.GetProcessorOptions(psr123.Guid, dsg1), "ProcessorOptions differ from expected Default");
```

Comment:
these assertions are duplicated many times in this file. I would make a helper method that takes in
1. The sut
2. The expected processor options
3. A list of PSR/DSG pairs

and the method can enumerate the list, asserting the expected options are returned from the call.